src/Discord.Net.Interactions/Info/Parameters/SlashCommandParameterInfo.cs

```diff
@@ -43,9 +55,14 @@ public class SlashCommandParameterInfo : CommandParameterInfo
         public bool IsAutocomplete { get; }
 
         /// <summary>
-        ///     Gets the Discord option type this parameter represents.
+        ///     Gets or sets whether this type should be treated as a complex parameter.
         /// </summary>
-        public ApplicationCommandOptionType DiscordOptionType => TypeConverter.GetDiscordType();
+        public bool IsComplexParameter { get; }
```

Comment:
Should this have a `set`?

Reply:
Users shouldnt be allowed to set this true without providing a valid object initializer delegate.